gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+		}
+	}
+
+	/** Test of the swap() method */
+	@Test
+	public void swapTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		longArray1.swap(1, 4);
+		Assert.assertArrayEquals(new long[] {1, 6, 4, 5, 3}, longArray1.toArray());
+		try {
+			longArray1.swap(100, 3);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+		try {
+			longArray1.swap(3, 100);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+	}
+
+	@Test
+	public void containsTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		Assert.assertTrue(longArray1.contains(3));
+		Assert.assertFalse(longArray1.contains(100));
+	}
+
+	/** Test of the hashCode() method */
+	@Test
+	public void hashCodeTest () {
+		/*
+		 * Testing if the hash code of an ordered long array is in fact the same provided as the hashCode() method from Object.
+		 */
+		Object longArrayUnordered = new LongArray(false, 16);
+		((LongArray)longArrayUnordered).add(2, 1, -6, 2);
+		((LongArray)longArrayUnordered).add(5, 50, 34, 42);
+		Assert.assertEquals(longArrayUnordered.hashCode(), ((LongArray)longArrayUnordered).hashCode());
+
+		LongArray longArrayOrdered = new LongArray();
+		longArrayOrdered.add(10, 20, 40, 50);
+		longArrayOrdered.add(100, 200, 400, 500);
+
+		/*
+		 * Expected hash calculation: 1. Start with h = 1 2. For each value in the array, compute (item ^ (item >>> 32)) and add it
+		 * to h as: h = h * 31 + (int)(item ^ (item >>> 32)) Detailed steps: - Iteration 1 (item = 10): h = 1 * 31 + 10 = 41 -
+		 * Iteration 2 (item = 20): h = 41 * 31 + 20 = 1 291 - Iteration 3 (item = 40): h = 1 291 * 31 + 40 = 40 061 - Iteration 4
+		 * (item = 50): h = 40 061 * 31 + 50 = 1 241 941 - Iteration 5 (item = 100): h = 1 241 941 * 31 + 100 = 38 500 271 -
+		 * Iteration 6 (item = 200): h = 38 500 271 * 31 + 200 = 1 193 508 601 - The next operation will overflow because h is an
+		 * int and can't be calculated if >= 2 147 483 647 (because negative numbers are defined by the first bit - The result of 1
+		 * 193 508 601 * 31 + 400 is 36 998 767 031 (1000 1001 1101 0100 1100 0110 0001 1011 0111) which is, truncated to 32 bits,
+		 * -1 655 938 633 (1001 1101 0100 1100 0110 0001 1011 0111) - Iteration 7 (item = 400): h = - 1 655 938 633 - Same here,
+		 * without redetailing the operation, we get, truncated to 32 bits, 205 510 429 - Iteration 8 (item = 500): h = 205 510 429
+		 *
+		 * Assert the calculated hash matches the expected value
+		 */
+
+		int expectedHash = 205510429;
+		Assert.assertEquals(expectedHash, longArrayOrdered.hashCode());
+	}
+
+	/** Test of the indexOf() method */
+	@Test
+	public void indexOfTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6, 6, 3, 9});
```

Comment:
It's a little odd that the fix was for lastIndexOf() taking a char parameter, when it should have been long, but all of the tested values would still be valid before the fix. If some were negative or greater than 65535, then this would be a better test of the change that was just made.